### PR TITLE
Formats

### LINKED ISSUE #212: Support sanger ab1 files
Related to #151 

Again, since there will be no data sync procedure for some users the pipeline should probably convert formats on-the-fly

### PR DESCRIPTION
**Pull request by averagehat**
Closes #212 #151

### CODE REVIEW

**Comment by averagehat** [comment]
This might break the functional tests because of the presence of the `ab1` files in `fixtures/functional`

File: ngs_mapper/file_formats.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from glob import glob
+import gzip
+from Bio import SeqIO
+
+{
+    'sff' : 'sff',
+    'ab1' : 'abi',  # SeqIO.convert(x, 'abi', y, 'fastq')
+}
+
+drop_ext = lambda s: '.'.join(s.split('.')[:-1])
```
Comment (necrolyte2) [inline ngs_mapper/file_formats.py]: I think split is fine, but it seems like you should be using os.path.splitext

Comment (averagehat) [inline ngs_mapper/file_formats.py]: I didn't know about that one. btw should we start using `path.py` in this project?

Comment (necrolyte2) [inline ngs_mapper/file_formats.py]: Probably a good idea. I have a slight reservation to introducing anything new right now, but that reservation is unfounded and probably due to old age.

What do you think?

Comment (averagehat) [inline ngs_mapper/file_formats.py]: *let's do it*

Comment (necrolyte2) [inline ngs_mapper/file_formats.py]: Haha, ok

**Comment by averagehat** [comment]
This, along with #199, are passing the second of the builds (which means the merge should also pass). Because this has been reviewed, I am going to merge it, and see what happens.

**Comment by averagehat** [comment]
Indeed, this merge succeeded happily

**Pull request merged**